packages/mobile/src/send/SendAmount/useTransactionCallbacks.ts

```diff
@@ -0,0 +1,207 @@
+import BigNumber from 'bignumber.js'
+import { useCallback, useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+import { hideAlert, showError } from 'src/alert/actions'
+import { RequestEvents, SendEvents } from 'src/analytics/Events'
+import { SendOrigin } from 'src/analytics/types'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { TokenTransactionType } from 'src/apollo/types'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import {
+  ALERT_BANNER_DURATION,
+  DEFAULT_DAILY_PAYMENT_LIMIT_CUSD,
+  DOLLAR_TRANSACTION_MIN_AMOUNT,
+} from 'src/config'
+import { getFeeEstimateDollars } from 'src/fees/selectors'
+import { AddressValidationType, secureSendPhoneNumberMappingSelector } from 'src/identity/reducer'
+import { getAddressValidationType } from 'src/identity/secureSend'
+import { RecipientVerificationStatus } from 'src/identity/types'
+import { convertToMaxSupportedPrecision } from 'src/localCurrency/convert'
+import {
+  useConvertBetweenCurrencies,
+  useLocalAmountInStableCurrency,
+  useStableCurrencyAmountInLocal,
+} from 'src/localCurrency/hooks'
+import {
+  getLocalCurrencyCode,
+  localCurrencyExchangeRateSelector,
+} from 'src/localCurrency/selectors'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { useRecipientVerificationStatus } from 'src/recipients/hooks'
+import { Recipient } from 'src/recipients/recipient'
+import useSelector from 'src/redux/useSelector'
+import { updateLastUsedCurrency } from 'src/send/actions'
+import { TransactionDataInput } from 'src/send/SendAmount'
+import { getFeeType, useDailyTransferLimitValidator } from 'src/send/utils'
+import { useCurrencyBalance } from 'src/stableToken/hooks'
+import { Currency } from 'src/utils/currencies'
+
+interface Props {
+  recipient: Recipient
+  localAmount: BigNumber
+  transferCurrency: Currency
+  origin: SendOrigin
+  isFromScan: boolean
+}
+
+function useTransactionCallbacks({
```

Comment:
Could you add a comment about what it does? like you added on the PR description.
I wonder if we should keep that helper in the same file to where it's used.

Reply:
The main file was very big, and scrolling up and down to see how things work was kind of tough. That was my reason for putting it in a separate file.